### PR TITLE
feat: add metrics endpoint for load balancers and servers

### PR DESCRIPTION
**Pull request by jooola**
Adds the missing metrics endpoint for the load balancers and servers resources.

- https://docs.hetzner.cloud/#load-balancers-get-metrics-for-a-loadbalancer
- https://docs.hetzner.cloud/#servers-get-metrics-for-a-server

### CODE REVIEW

File: hcloud/metrics/domain.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+TimeSeries = Dict[str, Dict[Literal["values"], List[Tuple[float, str]]]]
+
+
+class Metrics(BaseDomain):
+    """Metrics Domain
+
+    :param start: Start of period of metrics reported.
+    :param end: End of period of metrics reported.
+    :param step: Resolution of results in seconds.
+    :param time_series: Dict with time series data, using the name of time series as key.
```
Comment (apricote) [inline hcloud/metrics/domain.py:19]: Should we explain how this time series data is formatted inside the tuple? (`[ttimestamp, value]`)

**Review by apricote — APPROVED** [review]
Very nice!

File: hcloud/metrics/domain.py (hunk 1 above)
Comment (jooola) [reply hcloud/metrics/domain.py:19]: The full time series schema is overly complex, I would like to rely on the typing system to describe it. 

I could explain what the list of tuple are though!

Comment (apricote) [reply hcloud/metrics/domain.py:19]: Thats what I meant, full thing is pretty complex, but I feel the example in the API docs is quite good to get a grasp.

**Pull request merged**